src/processor/operator/recursive_extend/frontier_scanner.cpp

```diff
@@ -51,10 +51,15 @@ bool PathScanner::trailSemanticCheck(const std::vector<nodeID_t>&,
     const std::vector<relID_t>& edgeIDs) {
     common::rel_id_set_t set;
     for (auto i = 0u; i < edgeIDs.size() - 1; ++i) {
-        if (set.contains(edgeIDs[i])) {
+        internalID_t edgeID = edgeIDs[i];
+        if (RelIDMasker::needFlip(const_cast<internalID_t&>(edgeIDs[i]))) {
+            edgeID = RelIDMasker::getWithoutMark(const_cast<internalID_t&>(edgeIDs[i]));
+        } else {
+        }
```

Comment:
remove empty else statement so that the coding style is consistent in the codebase.